unblob/pool.py

```diff
@@ -85,18 +115,31 @@ def close(self):
     def _clear_input_queue(self):
         try:
             while True:
+                logger.warning("getnowait")
                 self._input.get_nowait()
         except queue.Empty:
             pass
 
     def _request_workers_to_quit(self):
         for _ in self._procs:
+            logger.warning("sentinel")
             self._input.put(_SENTINEL)
 
     def _wait_for_workers_to_quit(self):
         for p in self._procs:
+            logger.warning("join")
             p.join()
 
+        # Set back the original signal handlers
+        for sig, handler in self._original_signal_handlers.items():
+            signal.signal(sig, handler)
+
+        # Call the original signal handler with the fired and catched signal(s)
```

Comment:
Wait, we would call the original handler even for normal shutdown doesn't involving a signal as well...

Reply:
No, I don't think it would.

This iterates over the `self._signals_fired` list of the `MultiPool`, which is empty by default and elements are only added when a signal is caught. On a normal shutdown, this list is empty and nothing should happen.

Reply:
Ugh, totally missed that list :O